sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionClient.java

```diff
@@ -0,0 +1,219 @@
+package com.azure.core.http.httpurlconnection;
+
+import com.azure.core.http.*;
+import com.azure.core.http.httpurlconnection.implementation.HttpUrlConnectionResponse;
+import com.azure.core.util.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import com.azure.core.http.HttpHeader;
+
+import java.io.*;
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+
+/**
+ * This class provides a HttpUrlConnection implementation for the {@link HttpClient} interface.
+ * Creating an instance of this class can be achieved by using the {@link HttpUrlConnectionClientBuilder} class.
+ *
+ * @see HttpClient
+ * @see HttpUrlConnectionClientBuilder
+ */
+public class HttpUrlConnectionClient implements HttpClient {
+
+    private final ConcurrentHashMap<String, SocketClient> socketClientPool = new ConcurrentHashMap<>();
+
+    private final Duration connectionTimeout;
+    private final ProxyOptions proxyOptions;
+    private final Executor executor;
+    private final Configuration configuration;
+
+    public HttpUrlConnectionClient(Duration connectionTimeout, ProxyOptions proxyOptions, Executor executor, Configuration configuration) {
```

Comment:
This constructor shouldn't be public - it should be package-private, so that users are forced to use the builder